src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +28,36 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) obj;
+            return (color != null && color.equals(rectangle.color)
+                    || (rectangle.color == color))
+                    && (width != null && width.equals(rectangle.width)
+                    || (rectangle.width == width))
+                    && (length != null && length.equals(rectangle.length)
+                    || (rectangle.length == length));
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 0;
+        result = width != null ? HASHCODENUMBER * result + width.hashCode()
+                : HASHCODENUMBER * result;
```

Comment:
HASHCODENUMBER * result  will give us 0
is it supposed to be like that? maybe it's better to change result's initial value?

Reply:
There should have been "result +=", but I wrote a better version. Done)